drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/create_referral_job.rb

```diff
@@ -146,10 +146,9 @@ def setup_client_name(client, attrs)
       new_ccn.save!
 
       # ensure the ccn is the only primary
-      client.names.where.not(id: new_ccn.id).each do |ccn|
-        # update on each record for lifecycle hooks
-        ccn.update!(primary: false)
-      end
+      client.names.where.not(id: new_ccn.id).update_all(primary: false)
+      # propagate the primary name to the client record
+      client.assign_primary_name_fields
```

Comment:
Where does `client.save` get called? (I'm probably missing something, but `client.assign_primary_name_fields` doesn't seem to be persisting these fields.)

Reply:
Line 94 here https://github.com/greenriver/hmis-warehouse/blob/0850eb5fe207315020c91a04cb0cb9a6fa27cfcd/drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/create_referral_job.rb#L84-L94 and there is test coverage for it. I'll add a comment

Reply:
Thanks